### DISCUSSION TITLE
Add error message on `workflow.failures`

### DISCUSSION THREAD

**Discussion by sharadhirao** [argoproj/argo-workflows#10592] [Ideas]
Hi team,

We have been using argo workflows for the prod environment and developers don't have any permission to view the workflow UI because of security concerns. I am using google chat to get notifications using curl and workflow.failures parameter. We have integrated cosign and rekor for policy checking.
This `workflow.failures` provide just the exit code, not the exact error message.
Do we have any chance that we will get an exact error message instead of an exit code?
This will be a great help if you add this as a feature in future releases as we get an exact error instead of error codes as it would be easy to figure out what failed instead of going into workflow and checking.

Thanks in advance

**Comment by jswxstw** [comment]
https://github.com/argoproj/argo-workflows/blob/ea6dae9a30893632d863ff664803e17f3e9965f6/util/util.go#L51-L57

You can write the exact error message to `/dev/termination-log`, then it will be shown in `workflow.failures`.
This is a releated case: #13084
